src/app/item-review/d1-dtr-api-types.ts

```diff
@@ -0,0 +1,117 @@
+import { DtrReviewer, DimWorkingUserReview, DtrRating, DimUserReview } from "./dtr-api-types";
+
+/**
+ * A fetch request for a single weapon. Expected to be part of a bulk (array) fetch request.
+ */
+export interface D1ItemFetchRequest {
+  /** Reference ID for the weapon. */
+  referenceId: string;
+  /**
+   * The roll (available perks).
+   * Note that we only send random perks, so exotics, some raid weapons and other weapons don't pass this.
+   */
+  roll: string | null;
+}
+
+/** Bulk fetch response for a single weapon. */
+export interface D1ItemFetchResponse {
+  /** Reference ID for the weapon. */
+  referenceId: string;
+  /**
+   * The roll (available perks).
+   * Note that we only send random perks, so exotics, some raid weapons and other weapons don't pass this.
+   */
+  roll: string | null;
+  /** The rating from DTR. We use this. */
+  rating?: number;
+  /** The number of ratings that DTR has for the weapon (roll). */
+  ratingCount: number;
+  /** The number of highlighted ratings that DTR has for the weapon (roll). */
+  highlightedRatingCount: number;
+}
+
+/** A request for reviews for a single weapon. */
+export interface D1ItemReviewRequest {
+  /** Reference ID for the weapon. */
+  referenceId: string;
+  /**
+   * The roll (available perks).
+   * Note that we only send random perks, so exotics, some raid weapons and other weapons don't pass this.
+   */
+  roll: string | null;
+  /** What perks does the user have selected on it? */
+  selectedPerks: string | null;
+  /** What's the particular instance ID of the weapon? */
+  instanceId: string;
+}
+
+/** The local user's working rating. */
+export interface WorkingD1Rating extends DimWorkingUserReview {
```

Comment:
What makes a rating "working"?

Reply:
We let a user start on a review and hold it in cache until they submit it (and then treat it as though it were a review that had been received from the API).

Certainly open to suggestion on other names.